samples/matter/common/src/bridge/bridge_manager.cpp

```diff
@@ -176,6 +178,18 @@ CHIP_ERROR BridgeManager::AddSingleDevice(MatterBridgedDevice *device, BridgedDe
 {
 	uint8_t index{ 0 };
 	CHIP_ERROR err;
+	uint16_t duplicatedItemKeys[kMaxBridgedDevices];
+	BridgedDevicePair pair(device, dataProvider);
+
+	/* Check if the current provider is already contained in the map - if so, there is at least one duplicate */
+	bool isNewProvider = (Instance().mDevicesMap.GetDuplicates(pair, duplicatedItemKeys) == 0);
```

Comment:
Nit: do we need to allocate a variable for it? Maybe we could just move this check to the if statement two lines below.

Reply:
I wanted to achieve more self-documented code :) I'd like to keep it that way if you don't mind. Depending on the returned number of duplicates we may have different logic in the application, so I guess explicitness, even if redundant, does not hurt.